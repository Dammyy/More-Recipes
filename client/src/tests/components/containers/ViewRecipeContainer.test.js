import React from 'react';
import { shallow } from 'enzyme';
import { ViewRecipeContainer, mapStateToProps } from
  '../../../containers/ViewRecipeContainer';

describe('ViewRecipeContainer', () => {
  describe('Snapshot', () => {
    test('component should fully render', () => {
      const props = {
        recipe: {},
        recipesActions: { getRecipe: jest.fn(), },
        favoriteRecipe: jest.fn(),
        params: {},
        userId: 1,
        reviewsActions: { getReviews: jest.fn(), },
        reviews: [],
      };
      const component = shallow(<ViewRecipeContainer {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
  describe('container functions', () => {
    describe('mapStateToProps', () => {
      it('returns the expected properties', () => {
        const state = {
          getIn: () => ({ toJS: jest.fn() })
        };
        const mstp = mapStateToProps(state);
        expect(mstp).toHaveProperty('userId');
        expect(mstp).toHaveProperty('image');
        expect(mstp).toHaveProperty('recipe');
        expect(mstp).toHaveProperty('reviews');
      });
    });
  });

  describe('Instance methods', () => {
    describe('logout', () => {
      test('', () => {
        const props = {
          reviewsActions: { getReviews: jest.fn(), },
          recipesActions: {
            getRecipe: jest.fn(),
            getRecipeNoUserId: jest.fn(),
            favoriteRecipe: jest.fn(),
          },
          favoriteRecipe: jest.fn(),
          params: { id: 1 },
        };
        const component = shallow(<ViewRecipeContainer {...props} />);
        component.instance().favoriteRecipe();
        expect(props.reviewsActions.getReviews).toHaveBeenCalled();
        expect(props.reviewsActions.getReviews).toBeCalled();
      });
    });
  });
});
