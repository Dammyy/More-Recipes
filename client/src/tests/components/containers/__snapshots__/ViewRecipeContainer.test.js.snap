// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewRecipeContainer Snapshot component should fully render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewRecipeContainer
    favoriteRecipe={[MockFunction]}
    params={Object {}}
    recipe={Object {}}
    recipesActions={
      Object {
        "getRecipe": [MockFunction] {
          "calls": Array [
            Array [
              undefined,
              1,
            ],
          ],
        },
      }
    }
    reviews={Array []}
    reviewsActions={
      Object {
        "getReviews": [MockFunction] {
          "calls": Array [
            Array [
              undefined,
            ],
          ],
        },
      }
    }
    userId={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ViewRecipe
          details={undefined}
          favoriteRecipe={[Function]}
          id={undefined}
          image={undefined}
          recipe={Object {}}
          title={undefined}
          userId={1}
          voteRecipe={[Function]}
        />,
        <div
          id="reviews-form"
        >
          <Connect(AddReviewFormContainer)
            id={undefined}
          />
        </div>,
        <div
          className="reviews"
        >
          <Reviews
            id={undefined}
            reviews={Array []}
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "details": undefined,
          "favoriteRecipe": [Function],
          "id": undefined,
          "image": undefined,
          "recipe": Object {},
          "title": undefined,
          "userId": 1,
          "voteRecipe": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(AddReviewFormContainer)
            id={undefined}
          />,
          "id": "reviews-form",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Reviews
            id={undefined}
            reviews={Array []}
          />,
          "className": "reviews",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": undefined,
            "reviews": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ViewRecipe
            details={undefined}
            favoriteRecipe={[Function]}
            id={undefined}
            image={undefined}
            recipe={Object {}}
            title={undefined}
            userId={1}
            voteRecipe={[Function]}
          />,
          <div
            id="reviews-form"
          >
            <Connect(AddReviewFormContainer)
              id={undefined}
            />
          </div>,
          <div
            className="reviews"
          >
            <Reviews
              id={undefined}
              reviews={Array []}
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "details": undefined,
            "favoriteRecipe": [Function],
            "id": undefined,
            "image": undefined,
            "recipe": Object {},
            "title": undefined,
            "userId": 1,
            "voteRecipe": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(AddReviewFormContainer)
              id={undefined}
            />,
            "id": "reviews-form",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Reviews
              id={undefined}
              reviews={Array []}
            />,
            "className": "reviews",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": undefined,
              "reviews": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
